src/bitcoind.cpp

```diff
@@ -150,9 +159,20 @@ bool AppInit(int argc, char* argv[])
 
             pid_t sid = setsid();
             if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+                fprintf(stderr, "Error: setsid() failed: %s\n", strerror(errno));
+
+            (void)chdir("/");
```

Comment:
Sure, I'm not against changing the behavior in the future, we can do that later, one step at a time. However if you can do this PR as a simple change to `daemonize` that keeps the behavior the same, it's much easier to get it merged, even for 0.13.
If you start adding command-line flags this is too late for the feature freeze, so will only be considered after the 0.14 branch.